### PR TITLE
DEMOENG-1270 - Restructure template documentation

### PR DESCRIPTION
**Pull request by dremin**
### Summary

Complete overhaul of the documentation structure. Much of the existing content has been retained within the new structure, and there is a fair amount of new content as well.

You can view the updates locally by running `npm run start:docs`.

I believe there are still more opportunities for edits to the existing content, but I think this is already a large improvement!

### Checklist

- [x] Tested changes end to end
- [x] Updated documentation
- [x] Requested one or more reviewers

### CODE REVIEW

File: docs/docs/building/template-utilities/shared-sync-client.md
Comment (jhunter-twilio) [inline docs/docs/building/template-utilities/shared-sync-client.md:1]: do you think we could add a reference as an example to somewhere leveraging the the shared sync client?

File: docs/docs/04_package-overview.md
Comment (jhunter-twilio) [inline docs/docs/04_package-overview.md:1]: is this doc becoming redundant as we have similar information under "Building with the template > Getting Started"?

File: docs/docs/05_changelog.md
Comment (jhunter-twilio) [inline docs/docs/05_changelog.md:1]: should the change log have some updates in here about the TF adjustments for using serverless 😨

**Review by jhunter-twilio — CHANGES_REQUESTED** [review]
A huge step in the right direction - a couple minor additions and i think we can fold in these updates and iterate further - its a huge amount to bite off at once as it is, well done!

File: docs/docs/building/template-utilities/shared-sync-client.md
Comment (dremin) [reply docs/docs/building/template-utilities/shared-sync-client.md:1]: Done!

File: docs/docs/04_package-overview.md
Comment (dremin) [reply docs/docs/04_package-overview.md:1]: Yes, this doc is largely redundant, but I didn't have a good place to put the information that currently only exists on that page. Open to thoughts on where this info should go!

File: docs/docs/05_changelog.md
Comment (dremin) [reply docs/docs/05_changelog.md:1]: Merged PRs are our changelog, maybe we should get rid of the page?

File: docs/docs/04_package-overview.md
Comment (jhunter-twilio) [reply docs/docs/04_package-overview.md:1]: none at the moment, we can defer this decision to later

File: docs/docs/05_changelog.md
Comment (jhunter-twilio) [reply docs/docs/05_changelog.md:1]: that works for me!

Comment (dremin) [reply docs/docs/05_changelog.md:1]: Done!

**Review by jhunter-twilio — APPROVED** [review]

**Pull request merged**